### DISCUSSION TITLE
[ 功能建议 ] DDL和DML拆分成2个审核流程

### DISCUSSION THREAD

**Discussion by gddezero** [hhyo/Archery#1592] [Ideas]
### 目前遇到的问题/使用障碍
目前SQL上线审核和执行由各个业务开发lead来做，发生过几次大表执行ddl时将meta table锁死的事故

### 希望如何解决/实现它
希望将DDL审核和执行和DML拆分成2个审核流程，DDL由DBA审核执行，DML由业务开发lead来审核执行

### 其他信息
无

**Comment by gddezero** [comment]
这个功能的背景是DBA人手非常少，没有精力去审核执行所有的DDL和DML。这样的背景下，DBA主要把精力放在DDL的审核上，针对大表ddl，dba会使用pt-osc或ghost。很多开发没有这个意识，不清楚DDL可能会引起事故。
而DML可以由开发lead自行审核执行，毕竟开发最清楚DML的合理性和必要性。

**Comment by hhyo** [comment]
想法很好，不过容易解决一个问题又带来其他问题，可以考虑从权限下手，更细粒度的拆分执行权限

同时针对元数据锁goinception已经增加了限制，可以更新尝试，至少可以避免故障的产生

https://github.com/hanchuanchuan/goInception/pull/224
